apps/sms/views/conversation/test/unit/conversation_test.js

```diff
@@ -2551,6 +2552,67 @@ suite('conversation.js >', function() {
             });
           }).then(done, done);
         });
+
+        test('second chunk ready, then scroll up', function(done) {
+          var nbMessages = firstChunkSize + 2 * chunkSize;
+          for (var i = 1; i <= nbMessages; i++) {
+            MessageManager.getMessages.yieldTo(
+              'each', MockMessages.sms({ id: i })
+            );
+          }
+
+          TaskRunner.prototype.push.lastCall.returnValue.then(() => {
+            var messages = container.querySelectorAll('.message');
+
+            assert.lengthOf(messages, nbMessages);
+
+            Array.from(messages).forEach((message) => {
```

Comment:
__optional nit:__ I see you use the almost the same code 3 times, how about smth like this?
```
var messages = Array.from(container.querySelectorAll('.message'));

 ....

function assertMessagesVisibility(messages, lastVisibleMessageId) {
  messages.forEach((message) => {
    var id = +message.dataset.messageId;
    if (id <= lastVisibleMessageId) {
      assert.isFalse(
        message.classList.contains('hidden'),
        `Message ${id} should be displayed`
      );
    } else {
      assert.isTrue(
        message.classList.contains('hidden'),
        `Message ${id} should be hidden`
      );
    }
  });
}

.......
assertMessagesVisibility(messages, firstChunkSize);
.......
assertMessagesVisibility(messages, firstChunkSize + chunkSize);
.......
assertMessagesVisibility(messages, nbMessages);
```